src/IndexingWrapper.cpp

```diff
@@ -243,6 +245,71 @@ indexWrapper(Rcpp::CharacterVector & dbpath, Rcpp::DataFrame & dataFrame, Rcpp::
 	    }
 	}
 
+	//Add value slots
+	for (Rcpp::List::iterator it = slotList.begin(); it != slotList.end(); ++it) {
+	    Rcpp::List list = it.operator*();
+	    bool serialiseVal = false;
+	    bool dfVal = true; // use original row values in data frame
```

Comment:
Having said that, you could just replace the `dfVal` conditional with the `list.containsElementNamed(...)` expression you use to set `dfVal=false` at line 263.